### PR TITLE
Add backward test in LeNet and AlexNet

### PR DESCRIPTION
**Pull request by frankfzw**
## What changes were proposed in this pull request?

1. Add gradient points in `models` scripts to compute gradients in TensorFlow.
2. Change TensorflowLoader to accept `context` to store variables in BigDL for test
3. Move output transpose for LeNet into `testModel` function

 ## How was this patch tested?
 unit tests of LeNet and AlexNet

### CODE REVIEW

File: spark/dl/src/test/resources/tf/models/util.py [hunk 1]
```diff
@@ -57,27 +57,56 @@ def merge_checkpoint(input_graph,
...
 
     saver = tf.train.Saver()
     with tf.Session() as sess:
         init = tf.global_variables_initializer()
         sess.run(init) 
         sess.run(results)
+        sess.run(grad_results)
         saver.save(sess, output_path + '/model.chkp')
         tf.train.write_graph(sess.graph, output_path, 'model.pbtxt')
+        tf.summary.FileWriter(output_path + '/log', sess.graph)
```
Comment (yiheng) [inline spark/dl/src/test/resources/tf/models/util.py]: please comment this

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/tf/TensorflowLoaderSpec.scala [hunk 2]
```diff
@@ -282,8 +286,7 @@ class TensorflowLoaderSpec extends TensorflowSpecHelper{
   "Tensorflow lenet" should "be load correctly" in {
     testModel("lenet", Seq("LeNet/pool2/MaxPool:0"), true).foreach {
       case(tf, bigdl) =>
-        val transposed = bigdl.transpose(2, 3).transpose(3, 4)
-        tf.almostEqual(transposed, 1e-6) should be(true)
+        tf.almostEqual(bigdl, 1e-6) should be(true)
```
Comment (yiheng) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/tf/TensorflowLoaderSpec.scala]: don't we need the transpose here?

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/tf/TensorflowLoaderSpec.scala [hunk 3]
```diff
@@ -391,8 +397,50 @@ class TensorflowLoaderSpec extends TensorflowSpecHelper{
...
+        val t = tfNodes.asScala.filter(_.getName.contains(node.getName + "_grad"))(0)
+        t.getName ->
+          TensorflowToBigDL.toTensor(t.getAttrMap.get("value").getTensor, ByteOrder.LITTLE_ENDIAN)
+    }.toMap
+
+    val comparePair = tfOutputTensors.zip(bigdlOutputs).map{
+      x =>
+        val tensor = TensorflowToBigDL.toTensor(x._1, ByteOrder.LITTLE_ENDIAN)
+        if (transInput && tensor.dim() == 4) {
+          (tensor.transpose(4, 3).transpose(3, 2), x._2)
```
Comment (yiheng) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/tf/TensorflowLoaderSpec.scala]: what if the model is not a NHWC?
